perun-core/src/main/java/cz/metacentrum/perun/core/entry/AttributesManagerEntry.java

```diff
@@ -498,6 +498,32 @@ public AttributesManagerEntry() {
 		return attributes;
 	}
 
+	public List<Attribute> getAttributes(PerunSession sess, Resource resource, Group group, List<String> attrNames, boolean workWithGroupAttributes) throws PrivilegeException, InternalErrorException, ResourceNotExistsException, GroupNotExistsException, GroupResourceMismatchException, WrongAttributeAssignmentException {
+		Utils.checkPerunSession(sess);
+		getPerunBl().getResourcesManagerBl().checkResourceExists(sess, resource);
+		getPerunBl().getGroupsManagerBl().checkGroupExists(sess, group);
+		if(!getPerunBl().getGroupsManagerBl().getVo(sess, group).equals(getPerunBl().getResourcesManagerBl().getVo(sess, resource))) {
```

Comment:
There is exactly same check in Bl using private method from Bl. But if you still want to check, that both objects belong to the same VO, you can simply compare voId stored in both of them, like private method ``checkGroupIsFromTheSameVoLikeResource()`` in Bl does. There is no need to load objects from DB.